### PR TITLE
sys/board_common: add generic board_init() function

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Usually `board_init()` will just initialize the on-board LEDs.
This PR provides a generic, weak `board_init()` function that can be used as a fall-back if the board does not define a custom`board_init()` 

I also dropped the trivial `board_init()` functions from a few boards as an example. 

### Testing procedure

`examples/blinky` still works as expected.

### CODE REVIEW

File: sys/board_common/board_common.c [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * @}
+ */
+
+#include "board.h"
+#include "periph/gpio.h"
+#include "kernel_defines.h"
+
+#define LED_INIT(x) do {                                \
+                    gpio_init(LED##x##_PIN, GPIO_OUT);  \
+                    LED##x##_OFF;                       \
```
Comment (gschorcht) [inline sys/board_common/board_common.c:26]:
```suggestion
                        gpio_init(LED##x##_PIN, GPIO_OUT);  \
                        LED##x##_OFF;                       \
```

File: boards/esp32-ethernet-kit-v1_0/include/board.h [hunk 2]
```diff
@@ -25,14 +25,6 @@
 extern "C" {
 #endif
 
-/**
- * @brief Initialize the board specific hardware
- */
-static inline void board_init(void) {
-    /* there is nothing special to initialize on this board */
-    board_init_common();
```
Comment (gschorcht) [inline boards/esp32-ethernet-kit-v1_0/include/board.h:33]: `board_init_common` function should then be removed from `cpu/common/esp32/board_common.c`.

**Comment by fjmolinas** [comment]
ping

File: sys/board_common/board_common.c [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                    } while (0)
+
```
[the remark this answers is not in the document]
Comment (benpicco) [reply sys/board_common/board_common.c]: Sure can do - but that would not help me with the `gpio_init()`

File: boards/esp32-ethernet-kit-v1_0/include/board.h (hunk 2 above)
Comment (benpicco) [reply boards/esp32-ethernet-kit-v1_0/include/board.h:33]: It's still used by `esp32-ttgo-t-beam()`

File: sys/board_common/board_common.c [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+#endif
+#ifdef LED5_PIN
+    LED_INIT(5);
+#endif
+#ifdef LED6_PIN
+    LED_INIT(6);
+#endif
+#ifdef LED7_PIN
+    LED_INIT(7);
+#endif
```
Comment (benpicco) [inline sys/board_common/board_common.c]: Thinking about it, maybe this should better be moved to an `auto_init_leds`

Comment (maribu) [reply sys/board_common/board_common.c]: I like the idea :)

**Comment by fjmolinas** [comment]
This one now needs a rebase!

**Comment by fjmolinas** [comment]
The list could probably be even greatly reduced by adding `auto_init_buttons` and `auto_init_backlight`.

**Comment by benpicco** [comment]
Now this only gets rid of all the trivial / no-op `board_init()` functions - nice work!

**Review by fjmolinas — DISMISSED** [review]
ACK on my side let's wait and see if anyone has comments and merge this.

**Comment by leandrolanzieri** [comment]
Shouldn't we adapt documentation accordingly? It currently states:
> [...] Board initialization functions are defined in board.c. This file must at least define a `board_init()` function that is called at startup. [...]

File: boards/esp32-wrover-kit/board.c [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * @file
+ * @brief       Board specific definitions for esp32-wrover-kit
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ */
+
+#include "board.h"
+
+void board_init(void) {
```
Comment (aabadie) [inline boards/esp32-wrover-kit/board.c:21]: Style (ignore if you prefer):

```suggestion
void board_init(void)
{
```

**Comment by fjmolinas** [comment]
Hmm I see you also went for https://github.com/RIOT-OS/RIOT/pull/16013, I think that the last commit was a bit too greedy to do in this one, now there is the question if "kernel_init.h" is the right place for the definition, here it just seems like it was used as a header that is known to be included everywhere.

**Comment by benpicco** [comment]
Oh I just thought that if `board.c` no longer provides `board_init()`, there is no justification to declare it in `board.h`.

**Comment by fjmolinas** [comment]
> Oh I just thought that if board.c no longer provides 

I definitely agree, that's why in parallel I had nudged @chrysn on https://github.com/RIOT-OS/RIOT/pull/16013, it's just the placing of the function I'm unsure off. As the PR was before I would have merged tomorrow.

**Comment by benpicco** [comment]
What would be a better place for it?

**Comment by benpicco** [comment]
I dropped 48088cb01fb3da9e0064d4d2e41b7f34aa1175e0 for now.

**Comment by fjmolinas** [comment]
> What would be a better place for it?

I don't know... but everything that touches core is a sensible spot, which is why I would rather avoid it from this PR :)

**Review by fjmolinas — APPROVED** [review]
re-ACK

File: boards/esp32-ethernet-kit-v1_0/include/board.h (hunk 2 above)
Comment (gschorcht) [reply boards/esp32-ethernet-kit-v1_0/include/board.h:33]: > It's still used by `esp32-ttgo-t-beam()`

@benpicco In fact, `board_init_common()` in `cpu/common/esp32/board_common.c` is just empty. May I remove it?

Comment (benpicco) [reply boards/esp32-ethernet-kit-v1_0/include/board.h:33]: Sure!

**Pull request merged**